### DISCUSSION TITLE
Solution for error message "Can't set attribute x"

### DISCUSSION THREAD

**Discussion by m-guggenmos** [textext/textext#394] [General]
Just to share a little solution I found after trying to figure out why TexText didn't run for me. This is on Ubuntu 22.04, Inkscape 1.4-dev, TexText 1.9. I got the Error message `Can't set attribute x` whenever I wanted to save some input to the extension.

The culprit was in lines 402-403 in the file `base.py` of the extension:
```python
view_center.x = self.svg.uutounit(view_center.x, self.svg.unit)
view_center.y = self.svg.uutounit(view_center.y, self.svg.unit)
```

I found that setting `view_center` to a tuple instead of trying to set the attributes
```python
view_center = (self.svg.uutounit(view_center.x, self.svg.unit), 
               self.svg.uutounit(view_center.y, self.svg.unit))
```

resolved the error message and everything seemed to work.

**Comment by jcwinkler** [comment]
Thanks for the investigation for this upcoming Inkscape release. I will check how to make this run on both Inkscape extension APIs (in 1.3 and 1.4).
